babel.config.json

```diff
@@ -1,13 +1,16 @@
 {
```

Comment:
Do we need to keep **babel** only for test files? I suppose that we can use **ts-jest**. But it can be done in separate PR

Reply:
IIRC my implementation transpiles typescript using @babel/preset-typescript. This is generally recommended instead of using the Typescript compiler when Babel is used for the test of transpilation instead of combining methods.

If I'm wrong about that I'll double check when I have some dev time.

Tsc is used to generate d.ts files themselves, as babel only transpiles to javascript.

Reply:
Confirmed, this PR uses babel to transpile typescript for both build and test.